core/dbt/tracking.py

```diff
@@ -453,7 +454,14 @@ def track_run(run_command=None):
         track_invocation_end(invocation_context, result_type="ok")
     except (NotImplementedError, FailedToConnectError) as e:
         fire_event(MainEncounteredError(exc=str(e)))
+        if not isinstance(e, dbtException):
+            fire_event(MainStackTrace(stack_trace=traceback.format_exc()))
         track_invocation_end(invocation_context, result_type="error")
+    # import click.Abort as clickAbortError
+    # except clickAbortError: # a.k.a. MainKeyboardInterrupt
+    # if the logger isn't configured yet, it will use the default logger
+    #    fire_event(MainKeyboardInterrupt())  # how much do we care about this?
+    #    raise
```

Comment:
Trying to emulate this logic:
https://github.com/dbt-labs/dbt-core/blob/4d0ee2fc475cf7bc20caef20bb7a278a2c287c13/core/dbt/main.py#L142-L145

Not sure of exactly how this works now. The only thing we're losing out on is a structured event (`MainKeyboardInterrupt`) recording that a run was cancelled. Still, that might be cause enough to want to keep this logic around.